app/utils/smartapi/validator.py

```diff
@@ -43,3 +59,116 @@ def validate_symbol_and_get_token(
         raise SymbolNotFoundException(stock_symbol.split("-")[0])
 
     return all_symbols_data[stock_symbol], stock_symbol
+
+
+def check_market_open_between_dates(start_date: datetime, end_date: datetime):
+    """Checks whether the stock market is open between the given dates.
+
+    Parameters:
+    -----------
+    start_date: ``datetime``
+        Start date.
+    end_date: ``datetime``
+        End date.
+
+    Exceptions:
+    -----------
+    ``AllDaysHolidayException``
+        Raised when all days in the given date range are market holidays.
+    """
+    # Read the holidays data into list
+    holidays_data = read_text_data(NSE_HOLIDAYS_PATH)
+    # Check for any market open day between given dates.
+    # If you find any open day then definitely the data is not empty otherwise raise an error.
+    current_date = start_date
+    while current_date <= end_date:
+        index = bisect_left(holidays_data, current_date.strftime("%Y-%m-%d"))
+        if (
+            current_date.weekday() < 5
+            and index != len(holidays_data)
+            and holidays_data[index] != current_date.strftime("%Y-%m-%d")
+        ):
+            return
+        current_date += timedelta(days=1)
```

Comment:
Use `for` loop which is faster over `while` loop
![image](https://github.com/Munikumar09/market_data_api/assets/116535095/8fbfe049-a8ec-4989-a0a2-2fca4e92a05d)
[ref](https://www.pythonpool.com/for-vs-while-loop-python/)